metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -258,6 +331,62 @@ final class ReferenceProvider(
     }
   }
 
+  private def pcReferences(
+      path: AbsolutePath,
+      params: ReferenceParams,
+      forceIncludeLocalSearch: Boolean,
+      filterTargetFiles: AbsolutePath => Boolean = _ => true,
+  ): Future[List[ReferencesResult]] = {
+    val result = for {
+      name <- nameAtPosition(path, params.getPosition())
+      buildTarget <- buildTargets.inverseSources(path)
+      targetFiles = {
+        val localPath = Option.when(forceIncludeLocalSearch)(path)
+        pathsForName(buildTarget, localPath, name).filter(filterTargetFiles)
+      }
+      if targetFiles.nonEmpty
+    } yield compilers
+      .references(params, targetFiles, EmptyCancelToken)
```

Comment:
Would be great to forward an actual cancel token so if it takes longer to find references this can actually be cancelled.